fetch.bs

```diff
@@ -6710,6 +6752,417 @@ agent's <a>CORS-preflight cache</a> for which there is a <a>cache entry match</a
+       then decrement <var>quotaForOrigin</var> by <var>requestLength</var>.
+      </ol>
+    </ol>
+
+   <li>
+    <p>If <var>otherDocument</var>'s <a>container document</a> is a {{Document}} whose
+    <a for=Document>origin</a> is <a>same origin</a> with <var>requestClientDocument</var>'s
+    <a for=Document>origin</a>, then decrement <var>quota</var> by <var>otherNavigable</var>'s
+    <a>navigable container</a>'s <a>reserved deferred-fetch quota</a>.
+
+    <p class=note>Account for quota granted to child documents.
+  </ol>
+
+ <li><p>If <var>quota</var> is less than 0, then return 0.
+ <li><p>If <var>quota</var> is less than <var>quotaForOrigin</var>, then return <var>quota</var>.
+ <li><p>Return <var>quotaForOrigin</var>.
+</ol>
+</div>
+
+<div algorithm>
+<p>To <dfn export>reserve deferred-fetch quota</dfn> for a <a>navigable container</a>
+<var>container</var> given an <a for=/>origin</a> <var>originToNavigateTo</var>:
+
+<p class=note>This is called when <var>container</var> and the document that initiated the
+navigation (the "source document") are <a>same origin</a>. It potentially
+reserves either 64kb or 8kb of quota for the frame, if it is not <a>same origin</a>
+with its parent and the permissions policy allow. It is not observable to the cotnainer document
+whether the reserved quota was used in practice.
+This algorithm assumes that the container's document might delegate quota to the navigated frame,
+and the reserved quota would only apply in that case, and would be ignored if it ends up being
+shared.
+
+<ol>
+ <li><p>Set <var>container</var>'s <a>reserved deferred-fetch quota</a> to 0.
+
+ <li><p>If the <a data-lt="define an inherited policy for feature in container">inherited policy</a>
+ for "{{PermissionsPolicy/deferred-fetch}}", <var>container</var> and <var>originToNavigateTo</var>
+ is <code>Enabled</code>, and the <a>available deferred-fetch quota</a> for
+ <var>container</var>'s <a>container document</a> is equal or greater than
+ <a for="reserved deferred-fetch quota">normal quota</a>, then set <var>container</var>'s
+ <a>reserved deferred-fetch quota</a> to <a for="reserved deferred-fetch quota">normal quota</a> and
+ return.
+
+ <li><p>If the <a data-lt="define an inherited policy for feature in container">inherited policy</a>
+ for "{{PermissionsPolicy/deferred-fetch-minimal}}", <var>container</var> and
+ <var>originToNavigateTo</var> is <code>Disabled</code>, then return.
+
+ <li><p>If <var>container</var>'s <a>node document</a>'s <a for=Document>origin</a> is not
+ <a>same origin</a> with <var>container</var>'s <a>node navigable</a>'s
+ <a for=navigable>top-level traversable</a>'s <a>active document</a>'s <a for=Document>origin</a>,
+ then return.
+
+ <li><p>Let <var>containersWithReservedMinimalQuota</var> be <var>container</var>'s
+ <a>node navigable</a>'s <a for=navigable>top-level traversable</a>'s
+ <a>descendant navigables</a>, <a for=list data-lt=remove>removing</a> any <a for=/>navigable</a>
```

Comment:
```
removing any navigable whose reserved deferred-fetch quota is not minimal quota .
```

Does a `navigable` always have access to `reserved deferred-fetch quota`? Isn't the algo work only on `navigable container`?

Reply:
Yes, good catch, it's for the container